config/backend-config/backend-config.go

```diff
@@ -439,23 +276,53 @@ func Setup(configEnvHandler types.ConfigEnvI) (err error) {
 	return nil
 }
 
-func (bc *CommonBackendConfig) StartWithIDs(workspaces string) {
-	ctx, cancel := context.WithCancel(context.Background())
+func (bc *CommonBackendConfig) StartWithIDs(ctx context.Context, workspaces string) {
+	ctx, cancel := context.WithCancel(ctx)
 	bc.ctx = ctx
 	bc.cancel = cancel
 	bc.blockChan = make(chan struct{})
+	bc.waitForConfigErrs = make(chan error, 1)
 	rruntime.Go(func() {
-		pollConfigUpdate(ctx, bc.eb, workspaces)
+		bc.pollConfigUpdate(ctx, workspaces)
 		close(bc.blockChan)
 	})
 }
 
 func (bc *CommonBackendConfig) Stop() {
 	bc.cancel()
 	<-bc.blockChan
-	initializedLock.Lock()
-	initialized = false
-	initializedLock.Unlock()
+	bc.initializedLock.Lock()
+	bc.initialized = false
+	if bc.waitForConfigErrs != nil {
+		close(bc.waitForConfigErrs)
+	}
+	bc.initializedLock.Unlock()
+}
+
+// WaitForConfig waits until backend config has been initialized
+func (bc *CommonBackendConfig) WaitForConfig(ctx context.Context) error {
```

Comment:
`[question]` What the reason/motivation behind defining this method here ?

Reply:
It's always been there I think. I just moved it to the bottom next to `StartWithIDs` and `Stop` to make the file more readable and make more clear that they go together due to the fact that they use `bc.blockChan`, `bc.initialized` etcetera.